crates/core/src/expr/operator.rs

```diff
@@ -37,10 +37,14 @@ pub enum Operator {
 	AllEqual, // *=
 	AnyEqual, // ?=
 	//
-	Like,                      // ~
-	NotLike,                   // !~
-	AllLike,                   // *~
-	AnyLike,                   // ?~
+	#[revision(end = 3, convert_fn = "like_convert")]
+	Like, // ~
+	#[revision(end = 3, convert_fn = "like_convert")]
+	NotLike, // !~
+	#[revision(end = 3, convert_fn = "like_convert")]
+	AllLike, // *~
+	#[revision(end = 3, convert_fn = "like_convert")]
+	AnyLike, // ?~
```

Comment:
Just to be clear about the implications of this change. If someone has a DEFINE FUNCTION stored with a like operator, they will no longer be able to read that function from the DB after this change, right? They'll just get an error. So they'll need to re-define that function when they upgrade. Are there other places where `Operator` is likely to get written to disk? We should make sure this is well documented for the release and perhaps it makes sense to write a migration (or add a `fix`) that finds these situations and converts them to automatically use a similarity function.

Reply:
Aggreed.

I think the places where this can be an issue is the `DEFINE FUNCTION` and `DEFINE FIELD` with `VALUE <future> { .. }` or `DEFAULT <future> { .. }`. I don't believe it is possible to store the operator anywhere else.